core/sql/common/charinfo.cpp

```diff
@@ -693,3 +696,9 @@ Int32 CharInfo::getMaxConvertedLenInBytes(CharSet sourceCS,
   return ((sourceLenInBytes/minBytesPerChar(sourceCS)) *
           maxBytesPerChar(targetCS));
 }
+
+void CharInfo::initBuiltinCollationDB()
+{
+   builtinCollationDB_ = new CollationDB(NULL, mapCOArray, SIZEOF_CO);
```

Comment:
Is mapCOArray another global object? If it is const it is OK. If not, how do we know that it has been constructed already?

Reply:
Yes. C++ prologue ensures that it is initialized correctly. In master process, it will be initialized before dlOpen and in other processes it should be done before main is called.